cdap-authorization-ldap-role/src/main/java/io/cdap/cdap/security/authorization/ldap/role/RoleAuthorization.java

```diff
@@ -0,0 +1,292 @@
+      return entityIds;
+    }
+
+    PrincipalPermissions principalPermissions = getAllPermissions(username);
+
+    return entityIds.stream()
+      .filter(entity -> isVisible(entity, principalPermissions))
+      .collect(Collectors.toSet());
+  }
+
+  @Override
+  public void createRole(Role role) throws AccessException {
+    throw new AccessException("Method 'createRole' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void dropRole(Role role) throws AccessException {
+    throw new AccessException("Method 'dropRole' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void addRoleToPrincipal(Role role, Principal principal) throws AccessException {
+    throw new AccessException("Method 'addRoleToPrincipal' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void removeRoleFromPrincipal(Role role, Principal principal) throws AccessException {
+    throw new AccessException("Method 'removeRoleFromPrincipal' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public Set<Role> listRoles(Principal principal) throws AccessException {
+    throw new AccessException("Method 'listRoles' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public Set<Role> listAllRoles() throws AccessException {
+    throw new AccessException("Method 'listAllRoles' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void grant(Authorizable authorizable, Principal principal, Set<? extends Permission> set)
+    throws AccessException {
+    throw new AccessException("Method 'grant' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable, Principal principal, Set<? extends Permission> set)
+    throws AccessException {
+    throw new AccessException("Method 'revoke' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable) throws AccessException {
+    throw new AccessException("Method 'revoke' is not implemented, use external config file instead");
+  }
+
+  @Override
+  public Set<GrantedPermission> listGrants(Principal principal) throws AccessException {
+    throw new AccessException("Method 'listGrants' is not implemented, use external config file instead");
+  }
+
+  PrincipalPermissions getAllPermissions(String username) {
+    Set<String> groups = searcherService.searchGroups(username);
+    return roleProvider.getPrincipalPermissions(groups);
+  }
+
+  private void setSystemUser() {
+    if (!ignoreSystemUser) {
+      try {
+        UserGroupInformation info = UserGroupInformation.getCurrentUser();
+        systemUsername = info.getShortUserName();
+        LOG.info("System user: {}", systemUsername);
+      } catch (IOException e) {
+        ignoreSystemUser = true;
```

Comment:
why does it throw so generic exception?

Reply:
In hadoop code they wraps login exception to IOException
https://github.com/c9n/hadoop/blob/a315107901aebc8d0c2b9f530d191673d27922a2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java#L796